src/main/cpp/tests/cast_string.cpp

```diff
@@ -473,3 +481,138 @@ TEST_F(StringToDecimalTests, Edges)
+                                        "0001",
+                                        "0000.123",
+                                        "123",
+                                        "123.45",
+                                        "45.123",
+                                        "-45.123",
+                                        "0.45123",
+                                        "-0.45123",
+                                        "999999999999999999999",
+                                        "99999999999999999999",
+                                        "9999999999999999999",
+                                        "18446744073709551609",
+                                        "18446744073709551610",
+                                        "18446744073709551619999999999999",
+                                        "-18446744073709551609",
+                                        "-18446744073709551610",
+                                        "-184467440737095516199999999999997"};
+
+  std::vector<bool> valids{1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1};
+
+  auto expected =
+    cudf::strings::to_floats(strings_column_view(in), cudf::data_type{type_to_id<TypeParam>()});
+  expected->set_null_mask(cudf::test::detail::make_null_mask(valids.begin(), valids.end()));
+
+  auto const result = spark_rapids_jni::string_to_float(
+    data_type{type_to_id<TypeParam>()}, strings_column_view{in}, false, rmm::cuda_stream_default);
+
+  CUDF_TEST_EXPECT_COLUMNS_EQUAL(result->view(), expected->view());
+}
+
+TYPED_TEST(StringToFloatTests, InfNaN)
+{
+  cudf::test::strings_column_wrapper in{"NaN", "-Infinity", "inf", "Infinity", "-inf", "-nan"};
+
+  std::vector<bool> valids{1, 1, 1, 1, 1, 0};
+
+  auto expected =
+    cudf::strings::to_floats(strings_column_view(in), cudf::data_type{type_to_id<TypeParam>()});
+  expected->set_null_mask(cudf::test::detail::make_null_mask(valids.begin(), valids.end()));
+
+  auto const result = spark_rapids_jni::string_to_float(
+    data_type{type_to_id<TypeParam>()}, strings_column_view{in}, false, rmm::cuda_stream_default);
+
+  CUDF_TEST_EXPECT_COLUMNS_EQUAL(result->view(), expected->view());
+}
+
+TYPED_TEST(StringToFloatTests, InvalidValues)
+{
+  cudf::test::strings_column_wrapper in{"A", "null", "na7.62", "e", ".", "", "f"};
+
+  std::vector<bool> valids{0, 0, 0, 0, 0, 0, 0, 0};
+
+  auto expected =
+    cudf::strings::to_floats(strings_column_view(in), cudf::data_type{type_to_id<TypeParam>()});
+  expected->set_null_mask(cudf::test::detail::make_null_mask(valids.begin(), valids.end()));
+
+  auto const result = spark_rapids_jni::string_to_float(
+    data_type{type_to_id<TypeParam>()}, strings_column_view{in}, false, rmm::cuda_stream_default);
+
+  CUDF_TEST_EXPECT_COLUMNS_EQUAL(result->view(), expected->view());
+}
+
+TYPED_TEST(StringToFloatTests, ANSIInvalids)
+{
+  cudf::test::strings_column_wrapper in[] = {{"A"}, {"."}, {"e"}};
```

Comment:
I think that we should add more cases for the ansi test. Here are only very simple cases.

Reply:
I added tests from the plugin that failed to ensure they didn't regress. The plugin tests are VERY extensive.